README.md
# ManHunt
The ManHunt Project. Write it as ManHunt not manhunt or Manhunt, capitalization counts for me


Releases.txt
Food and Hunger Bar in the next release (Release V2)
Potions in a newer release (Release V3)

0.0.0.1:
    Finished writing this frickin script or smth (Releases.txt)
0.0.0.1.1:
    improve my spellin. or not? (Releases.txt)
0.0.0.1.2:
    put releases in correct order for better known goals
0.0.1.1.1:
    Finished writing script to generate random spawn building (Generate_Spawn_Building.py)
0.0.1.1.4:
    Finished writing script to generate nether portal (Generate_Nether_Portal_Building.py)
0.0.1.1.8:
    Finished writing script to generate end portal letter (Generate_End_Portal_Letter.py)
0.0.1.2:
    Finished measuring distances between all buildings (Building_Distances.csv)
0.0.1.3:
    Finished writing script to "Smartly" generate Spawn (Generate_Spawn_Building_Smartly.py)
0.0.1.3.12:
    Finished writing script to "Smartly" generate Nether Portal (Generate_Nether_Portal_Building_Smartly.py)
0.0.1.3.32:
    Finished writing script to "Smartly" generate End Portal (Generate_End_Portal_Building_Smartly.py)
0.0.1.4.1:
    Finished writing script recording video part of "Record" folder (Record_Video.py)
0.0.1.4.2:
    Finished writing script recording audio part of the "Record" folder (Record_Audio.py)
0.0.1.4.3:
    Finished writing script both recording parts of the "Record" folder (Record_Footage.py)
0.0.1.5:
    Finished writing script for Swords to send damage [Script for Swords] (Steb_lol.py)
0.0.1.5.10:
    Finished writing script for Vest to recieve Stab/Sword damage [Script for Vest] (Pian_Sword.py)
0.0.1.6:
    Finished writing script for Axe to send damage [Script for Swords] (Chop_lol.py)
0.0.1.6.10:
    Finished writing script for Vest to recieve Axe damage [Script for Vest] (Pian_Axe.py)
0.0.1.7:
    Finished writing script for Vest to Recieve damage [Script for Vest] (Le_Fist.py)
0.0.1.8:
    Finished writing script for healing over time (Le_Paower_of_Regeneration.py)
0.0.1.9.1:
    Finished writing server script for Texting [Jimmy's Part] {HTML/CSS part} (txt.css)
0.0.1.9.2:
    Started writing server script for Calling [Jimmy's Part] {The Basics} (Insult.css)
0.0.1.9.2.5:
    Improved server script for Calling [Jimmy's Part] {With mute and deafen} (Insult.css)
0.0.1.9.2.10:
    Finished writing server script for Calling [Jimmy's Part] {Finalized it} (Insult.css)
0.0.1.9.3:
    Made GUI for Texting and Calling Program [Jimmy's Part] {Made the basics} (GUI.css) lol try to pronounce the extention lol
0.0.1.9.3:
    Made GUI for Texting and Calling Program [Jimmy's Part] {Made it 𝒷𝑒𝒶𝓊𝓉𝒾𝒻𝓊𝓁} (GUI.css)
0.0.1.9.3:
    Made GUI for Texting and Calling Program [Jimmy's Part] {FINALIZED IT} (GUI.css)
0.0.1.10:
    Created script for son drones/monitors
0.0.1.10.1:
    Created script for mother drone/monitor
0.0.1.10.1.1:
    Tested time for message full ping
0.0.1.10.2:
    Betterfied All scripts
0.0.1.10.3:
    Finished script for location recording
0.0.1.10.4:
    Finished script for location detection
0.0.1.10.5:
    Finished PCB design for drone and bought the Pieces
0.0.1.10.6:
    Started assembling the parts of drone
0.0.1.10.7:
    Made it successfully fly
0.0.1.10.8:
    Stablized it
0.0.1.10.9:
    Found Bugs and stomped them
0.0.1.10.10
    Finalized it
0.0.1.11.1:
    Finished creating the Minecraft-like GUI [Healthbar]
0.0.1.11.2:
    Finished creating the Minecraft-like GUI [HotBar]
0.0.1.11.3:
    Finished creating the Minecraft-like GUIs [The rest small details\
    such as:
        Aiming thing, like y'know the thing that is always in the middle?, help my vocab or keep it like this if you like my way of taiping or somthing
        Hit effect
        etc. I'll get back when I think of it
0.0.1.11.4:
    Finalized Minecraft GUIs [Well, Everything]
0.0.1.11.5:
    Started Making GUIs work [Like respond to punches and stuff, y'know]
0.0.1.11.6:
    Finalized GUI [Made everything work perfectly]
0.0.1.12.1:
    Started on tootaly not stealing Minecraft's original sounds and soundtracks lol.
0.0.1.13:
    Finished Every needed stuff in Vest_Main
0.0.1.14:
    Crushed all bugs with a tank
0.0.1.15:
    Finalized everything with Vest_Main


Programing/Vest_Main/Main.py
'''The Main Program'''


'''Starting up and importing modules'''

print("Starting")

'''This indicates to the user that the program/code 
is running and (hopefully) working'''

import time
'''This imports time module'''
time.sleep(0.5)
'''This adds a delay so the user notices'''
print("Starting.")
'''This indicates that the program has entered 
the second stage of starting up'''

import random
'''This imports the random module 
(You can add more modules if it is needed)'''
time.sleep(0.05)
print("Starting..")

import cv2 as cv
'''This Imports the OpenCV Module for Computer Vision'''
time.sleep(0.05)
print("Starting...")

time.sleep(0.05)
print("Finished Loading.")
'''This tells the user that the program 
has finished loading modules'''


'''Create Variables'''

'''The code below will create the neccessery variables
You can add some variables if you want or is needed'''


'''The code for certain destinations of the ("minecraft") manhunt'''

Building_Letters = ("A", "B", "B2", "C", "E", "E2", "F", "G", "H1", "H", "K", "L", "M", "N", "N2", "P")

'''These are all the building letters that exist in Henry's Complex'''

All_Buildings = \
("A:32:1",
 "B:30:1", "B:31:1", "B:35:1", "B:36:1", "B:35:2", "B:36:2",
 "C:27", "C:28",
 "E:26:1", "E:27:1",
 "E:1:2", "E:2:2",
 "F:34:1",
 "G:29:1",
 "H1:1:1", "H1:2:1", "H1:3:1",
 "H:6:1", "H:7:1",
 "K:1:1", "K:2:1", "K:3:1", "K:4:1", "K:5:1", "K:6:1", "K:7:1",
 "L:1:1",
 "M:1:1", "M:2:1",
 "N:1:1", "N:2:1",
 "N:1:2", "N:2:2",
 "P:1:1", "P:2:1", "P:3:1", "P:4:1", "P:5:1",
 "P:6:1", "P:7:1", "P:8:1", "P:9:1", "P:10:1", "P:11:1")

All_Buildings_Dictionary = {\
        "A":{\
            "1":[32]}, \
        "B":{\
            "1":["30", "31", "35", "36"], \
            "2":["35", "36"]}, \
        "C":{\
            "1":["27", "28"]}, \
        "E":{\
            "1":["26", "27"], \
            "2":["1", "2"]}, \
        "F":{\
            "1":["34"]}, \
        "G":{\
            "1":["29"]}, \
        "H1":{\
            "1":["1", "2", "3"]}, \
        "H":{\
            "1":["6", "7"]}, \
        "K":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}, \
        "L":{\
            "1":["1"]}, \
        "M":{\
            "1":["1", "2"]}, \
        "N":{\
            "1":["1", "2"], \
            "2":["1", "2"]}, \
        "P":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}}



Underground_Letters = []

'''This is all the Letters of the underground parking lot.'''

Underground_Numbers = []

'''This gives the computer a better advantage of how to better sense where 
the other objects (Blaze rods) are.'''

Spawn = None

'''This is where the playeruser starts the 
speed run, it is usually a building, 
 a building has a letter and a number, the spawn 
 area specifically can be 
not chosen, chosen smartly or chosen randomly'''

Time_Of_Nether_Portal_Spawn = None
'''variable that determines the time till the nether portal spawns'''

'''This is the amount of time that the player 
has to survive before the Nether Portal
Spawns, if the Player didn't manage to survive in this 
amount of time, the game ends or the
play has lost, the time is 2 to 5 minutes'''

Nether_Portal_First_Letter = None

'''This is the first letter of the location of the nether portal, 
every player needs to go through this building in 
order to continue to hunt down the speedrunner '''

Nether_Portal_First_Number = None

'''This is the first number of the location of the nether portal, 
every player needs to go through this building in 
order to continue to hunt down the speedrunner '''

Nether_Portal = None

'''This is the building where all the players 
(including the manhunters) have to go through 
in order to continue the game.
the building name is created by combining two 
values together to create the random building'''

'''To indicate whether the speedrunner has gone 
through the nether portal, there will be a device in the gates 
of the elevator, hidden to not be caught by normal people and stolen
there will be a device in ever building because the Nether Portals 
are going to be chosen randomly'''

Nether_Fortress = None

'''This is the area where there is blaze rods, which is required to 
access the end portal, it gives only a letter and the blaze rods will
be nowhere out of that area. When the sensors of the Raspberry pi on the 
speedrunner detects the 12 blaze rods it will give the speedrunner the 
compass navigation (similar to minecraft compasses) but it refreshes 
when the player inputs it to the Raspberry Pi like how eyes of ender work.'''

# More about blaze rods and how they work.
'''This is the place where the blaze rods are found, aka an small 
electronic device, that indicates if the blaze rod is found when
pressed a button on it, the speedrunner needs to collect a total of 
12 blaze rods hidden in the parking lot, at the end of the game, they
will be picked up using location and bluetooth and sound. There are 
going to be 12 blaze rods not in every part of the parking lot
otherwise, it would first, cost a lot of money, or secondly, 
another blaze rod that was in another area of the parking lot, be
taken by the speedrunner which isn't supposed to happen. 
The Nether Fortresses are going to be very far away from each other
to prevent getting take by a closer place. The blaze rods will be hidden,
but will be more like the eyes of ender, you can tell the Raspberry Pi 
show the directions like a compass but it can only be used 36 times 
to tell the directions, 3 times per rod. If all of the chances to 
tell the location the speedrunner may either, give up on life, or keep 
on finding it with bare eyes. '''

StrongHold = None

'''This is the most important part of the game, and the most complex, 
The speedrunner has to both escape from being killed by the hunters 
and also find the stronghold, there will be devices like the ones in the
Nether portal to indicate, like the nether portal, the device will be also 
put into every building or for the cheaper choice, it can be instead just 
be the Nether Portal sensor but with more code. The program will specify the
building letter, but won't tell the numbers, the speedrunner then has to 
go through all of the buildings with that letter and then there is going 
to be a status bar showing how close the speerunner is to the portal,
this feature of showing how close (or hot) he is to the end can be removed'''


Programing/Vest_Main/Create_Variables.py
'''The code below will create the neccessery variables
You can add some variables if you want or is needed'''


'''The code for certain destinations of the ("minecraft") manhunt'''

Building_Letters = ("A", "B", "B2", "C", "E", "E2", "F", "G", "H1", "H", "K", "L", "M", "N", "N2", "P")

'''These are all the building letters that exist in Henry's Complex'''

All_Buildings = \
("A:32:1",
 "B:30:1", "B:31:1", "B:35:1", "B:36:1", "B:35:2", "B:36:2",
 "C:27", "C:28",
 "E:26:1", "E:27:1",
 "E:1:2", "E:2:2",
 "F:34:1",
 "G:29:1",
 "H1:1:1", "H1:2:1", "H1:3:1",
 "H:6:1", "H:7:1",
 "K:1:1", "K:2:1", "K:3:1", "K:4:1", "K:5:1", "K:6:1", "K:7:1",
 "L:1:1",
 "M:1:1", "M:2:1",
 "N:1:1", "N:2:1",
 "N:1:2", "N:2:2",
 "P:1:1", "P:2:1", "P:3:1", "P:4:1", "P:5:1",
 "P:6:1", "P:7:1", "P:8:1", "P:9:1", "P:10:1", "P:11:1")

All_Buildings_Dictionary = {\
        "A":{\
            "1":[32]}, \
        "B":{\
            "1":["30", "31", "35", "36"], \
            "2":["35", "36"]}, \
        "C":{\
            "1":["27", "28"]}, \
        "E":{\
            "1":["26", "27"], \
            "2":["1", "2"]}, \
        "F":{\
            "1":["34"]}, \
        "G":{\
            "1":["29"]}, \
        "H1":{\
            "1":["1", "2", "3"]}, \
        "H":{\
            "1":["6", "7"]}, \
        "K":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}, \
        "L":{\
            "1":["1"]}, \
        "M":{\
            "1":["1", "2"]}, \
        "N":{\
            "1":["1", "2"], \
            "2":["1", "2"]}, \
        "P":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}}



Underground_Letters = []

'''This is all the Letters of the underground parking lot.'''

Underground_Numbers = []

'''This gives the computer a better advantage of how to better sense where 
the other objects (Blaze rods) are.'''

Spawn = None

'''This is where the playeruser starts the 
speed run, it is usually a building, 
 a building has a letter and a number, the spawn 
 area specifically can be 
not chosen, chosen smartly or chosen randomly'''

Time_Of_Nether_Portal_Spawn = None
'''variable that determines the time till the nether portal spawns'''

'''This is the amount of time that the player 
has to survive before the Nether Portal
Spawns, if the Player didn't manage to survive in this 
amount of time, the game ends or the
play has lost, the time is 2 to 5 minutes'''

Nether_Portal_First_Letter = None

'''This is the first letter of the location of the nether portal, 
every player needs to go through this building in 
order to continue to hunt down the speedrunner '''

Nether_Portal_First_Number = None

'''This is the first number of the location of the nether portal, 
every player needs to go through this building in 
order to continue to hunt down the speedrunner '''

Nether_Portal = None

'''This is the building where all the players 
(including the manhunters) have to go through 
in order to continue the game.
the building name is created by combining two 
values together to create the random building'''

'''To indicate whether the speedrunner has gone 
through the nether portal, there will be a device in the gates 
of the elevator, hidden to not be caught by normal people and stolen
there will be a device in ever building because the Nether Portals 
are going to be chosen randomly'''

Nether_Fortress = None

'''This is the area where there is blaze rods, which is required to 
access the end portal, it gives only a letter and the blaze rods will
be nowhere out of that area. When the sensors of the Raspberry pi on the 
speedrunner detects the 12 blaze rods it will give the speedrunner the 
compass navigation (similar to minecraft compasses) but it refreshes 
when the player inputs it to the Raspberry Pi like how eyes of ender work.'''

# More about blaze rods and how they work.
'''This is the place where the blaze rods are found, aka an small 
electronic device, that indicates if the blaze rod is found when
pressed a button on it, the speedrunner needs to collect a total of 
12 blaze rods hidden in the parking lot, at the end of the game, they
will be picked up using location and bluetooth and sound. There are 
going to be 12 blaze rods not in every part of the parking lot
otherwise, it would first, cost a lot of money, or secondly, 
another blaze rod that was in another area of the parking lot, be
taken by the speedrunner which isn't supposed to happen. 
The Nether Fortresses are going to be very far away from each other
to prevent getting take by a closer place. The blaze rods will be hidden,
but will be more like the eyes of ender, you can tell the Raspberry Pi 
show the directions like a compass but it can only be used 36 times 
to tell the directions, 3 times per rod. If all of the chances to 
tell the location the speedrunner may either, give up on life, or keep 
on finding it with bare eyes. '''

StrongHold = None

'''This is the most important part of the game, and the most complex, 
The speedrunner has to both escape from being killed by the hunters 
and also find the stronghold, there will be devices like the ones in the
Nether portal to indicate, like the nether portal, the device will be also 
put into every building or for the cheaper choice, it can be instead just 
be the Nether Portal sensor but with more code. The program will specify the
building letter, but won't tell the numbers, the speedrunner then has to 
go through all of the buildings with that letter and then there is going 
to be a status bar showing how close the speerunner is to the portal,
this feature of showing how close (or hot) he is to the end can be removed'''


Programing/Vest_Main/Shuffle_Variables.py
'''This Program Randomly Shuffles the variables'''

'''I copied from the original program with some simple modifications to make it more clear'''

'''The code below will create the neccessery variables
You can add some variables if you want or is needed'''

import random


'''The code for certain destinations of the ("minecraft") manhunt'''

Building_Letters = ("A", "B", "B2", "C", "E", "E2", "F", "G", "H1", "H", "K", "L", "M", "N", "N2", "P")

'''These are all the building letters that exist in Henry's Complex'''

All_Buildings = \
("A:32:1",
 "B:30:1", "B:31:1", "B:35:1", "B:36:1", "B:35:2", "B:36:2",
 "C:27:1", "C:28:1",
 "E:26:1", "E:27:1",
 "E:1:2", "E:2:2",
 "F:34:1",
 "G:29:1",
 "H1:1:1", "H1:2:1", "H1:3:1",
 "H:6:1", "H:7:1",
 "K:1:1", "K:2:1", "K:3:1", "K:4:1", "K:5:1", "K:6:1", "K:7:1",
 "L:1:1",
 "M:1:1", "M:2:1",
 "N:1:1", "N:2:1",
 "N:1:2", "N:2:2",
 "P:1:1", "P:2:1", "P:3:1", "P:4:1", "P:5:1",
 "P:6:1", "P:7:1", "P:8:1", "P:9:1", "P:10:1", "P:11:1")

All_Buildings_Dictionary = {\
        "A":{\
            "1":["32"]}, \
        "B":{\
            "1":["30", "31", "35", "36"], \
            "2":["35", "36"]}, \
        "C":{\
            "1":["27", "28"]}, \
        "E":{\
            "1":["26", "27"], \
            "2":["1", "2"]}, \
        "F":{\
            "1":["34"]}, \
        "G":{\
            "1":["29"]}, \
        "H1":{\
            "1":["1", "2", "3"]}, \
        "H":{\
            "1":["6", "7"]}, \
        "K":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}, \
        "L":{\
            "1":["1"]}, \
        "M":{\
            "1":["1", "2"]}, \
        "N":{\
            "1":["1", "2"], \
            "2":["1", "2"]}, \
        "P":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}}

Numbers_Corresponding_To_Building_Letters = {"1":"A", "2":"B", "3":"C", "4":"E", "5":"F", "6":"G", "7":"H1", "8":"H", "9":"K", "10":"L", "11":"M", "12":"N", "13":"P"}
'''This way I can use the Random module to generate random building letters'''
Spawn_Letter = str(Numbers_Corresponding_To_Building_Letters[str(random.randint(1,12))])
'''Write this down in a variable so that I can display it as well'''
Spawn_First_Or_Second_Building = str(random.randint(1,2))
'''Because there are so  many buildings in Henry's complex with the same letters, I have split it into either 1 or 2 this variable is randomly generated either 1 or 2'''
Spawn_Building_Number = 0

for key in All_Buildings_Dictionary:
    if Spawn_Letter == "A" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = "32"
        Spawn = All_Buildings[0]
        break
    if Spawn_Letter == "B" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number_With_Skips = str(random.randint(1, 2))
        if Spawn_Building_Number_With_Skips == "1":
            Spawn_Building_Number = str(random.randint(30, 31))
            if Spawn_Building_Number == "30":
                Spawn = All_Buildings[1]
                break
            if Spawn_Building_Number == "31":
                Spawn = All_Buildings[2]
                break
            break
        if Spawn_Building_Number_With_Skips == "2":
            Spawn_Building_Number = str(random.randint(35, 36))
            if Spawn_Building_Number == "35":
                Spawn = All_Buildings[3]
                break
            if Spawn_Building_Number == "36":
                Spawn = All_Buildings[4]
                break
            break
        break
    if Spawn_Letter == "B" and Spawn_First_Or_Second_Building == "2":
        Spawn_Building_Number = str(random.randint(35, 36))
        if Spawn_Building_Number == "35":
            Spawn = All_Buildings[5]
            break
        if Spawn_Building_Number == "36":
            Spawn = All_Buildings[6]
            break
        break
    if Spawn_Letter == "C" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(27, 28))
        if Spawn_Building_Number == "27":
            Spawn = All_Buildings[7]
            break
        if Spawn_Building_Number == "28":
            Spawn = All_Buildings[8]
            break
        break
    if Spawn_Letter == "E" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(26, 27))
        if Spawn_Building_Number == "26":
            Spawn = All_Buildings[9]
            break
        if Spawn_Building_Number == "27":
            Spawn = All_Buildings[10]
            break
        break
    if Spawn_Letter == "E" and Spawn_First_Or_Second_Building == "2":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[11]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[12]
            break
        break
    if Spawn_Letter == "F" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = "34"
        Spawn = All_Buildings[13]
        break
    if Spawn_Letter == "G" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = "29"
        Spawn = All_Buildings[14]
        break
    if Spawn_Letter == "H1" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 3))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[15]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[16]
            break
        if Spawn_Building_Number == "3":
            Spawn = All_Buildings[17]
            break
        break
    if Spawn_Letter == "H" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(6, 7))
        if Spawn_Building_Number == "6":
            Spawn = All_Buildings[18]
            break
        if Spawn_Building_Number == "7":
            Spawn = All_Buildings[19]
            break
        break
    if Spawn_Letter == "K" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 7))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[20]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[21]
            break
        if Spawn_Building_Number == "3":
            Spawn = All_Buildings[22]
            break
        if Spawn_Building_Number == "4":
            Spawn = All_Buildings[23]
            break
        if Spawn_Building_Number == "5":
            Spawn = All_Buildings[24]
            break
        if Spawn_Building_Number == "6":
            Spawn = All_Buildings[25]
            break
        if Spawn_Building_Number == "7":
            Spawn = All_Buildings[26]
            break
        break
    if Spawn_Letter == "L" and Spawn_First_Or_Second_Building == "1":
        Spawn = All_Buildings[27]
        Spawn_Building_Number = "1"
        break
    if Spawn_Letter == "M" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[28]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[29]
            break
        break
    if Spawn_Letter == "N" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[30]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[31]
            break
        break
    if Spawn_Letter == "N" and Spawn_First_Or_Second_Building == "2":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[32]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[33]
            break
        break
    if Spawn_Letter == "P" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 11))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[34]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[35]
            break
        if Spawn_Building_Number == "3":
            Spawn = All_Buildings[36]
            break
        if Spawn_Building_Number == "4":
            Spawn = All_Buildings[37]
            break
        if Spawn_Building_Number == "5":
            Spawn = All_Buildings[38]
            break
        if Spawn_Building_Number == "6":
            Spawn = All_Buildings[39]
            break
        if Spawn_Building_Number == "7":
            Spawn = All_Buildings[40]
            break
        if Spawn_Building_Number == "8":
            Spawn = All_Buildings[41]
            break
        if Spawn_Building_Number == "9":
            Spawn = All_Buildings[42]
            break
        if Spawn_Building_Number == "10":
            Spawn = All_Buildings[43]
            break
        if Spawn_Building_Number == "11":
            Spawn = All_Buildings[44]
            break
        break
    break
print('Spawn Letter: "', str(Spawn_Letter), '", First Or Second Building: "', str(Spawn_First_Or_Second_Building), '" ', "Building Number: ", str(Spawn_Building_Number), sep="")

Programing/Vest_Main/Generate_Spawn_Building.py
'''This Program Randomly Shuffles the variables'''

'''I copied from the original program with some simple modifications to make it more clear'''

'''The code below will create the neccessery variables
You can add some variables if you want or is needed'''

import random


'''The code for certain destinations of the ("minecraft") manhunt'''

Building_Letters = ("A", "B", "B2", "C", "E", "E2", "F", "G", "H1", "H", "K", "L", "M", "N", "N2", "P")

'''These are all the building letters that exist in Henry's Complex'''

All_Buildings = \
("A:32:1",
 "B:30:1", "B:31:1", "B:35:1", "B:36:1", "B:35:2", "B:36:2",
 "C:27:1", "C:28:1",
 "E:26:1", "E:27:1",
 "E:1:2", "E:2:2",
 "F:34:1",
 "G:29:1",
 "H1:1:1", "H1:2:1", "H1:3:1",
 "H:6:1", "H:7:1",
 "K:1:1", "K:2:1", "K:3:1", "K:4:1", "K:5:1", "K:6:1", "K:7:1",
 "L:1:1",
 "M:1:1", "M:2:1",
 "N:1:1", "N:2:1",
 "N:1:2", "N:2:2",
 "P:1:1", "P:2:1", "P:3:1", "P:4:1", "P:5:1",
 "P:6:1", "P:7:1", "P:8:1", "P:9:1", "P:10:1", "P:11:1")

All_Buildings_Dictionary = {\
        "A":{\
            "1":["32"]}, \
        "B":{\
            "1":["30", "31", "35", "36"], \
            "2":["35", "36"]}, \
        "C":{\
            "1":["27", "28"]}, \
        "E":{\
            "1":["26", "27"], \
            "2":["1", "2"]}, \
        "F":{\
            "1":["34"]}, \
        "G":{\
            "1":["29"]}, \
        "H1":{\
            "1":["1", "2", "3"]}, \
        "H":{\
            "1":["6", "7"]}, \
        "K":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}, \
        "L":{\
            "1":["1"]}, \
        "M":{\
            "1":["1", "2"]}, \
        "N":{\
            "1":["1", "2"], \
            "2":["1", "2"]}, \
        "P":{\
            "1":["1", "2", "3", "4", "5", "6", "7"]}}

Numbers_Corresponding_To_Building_Letters = {"1":"A", "2":"B", "3":"C", "4":"E", "5":"F", "6":"G", "7":"H1", "8":"H", "9":"K", "10":"L", "11":"M", "12":"N", "13":"P"}
'''This way I can use the Random module to generate random building letters'''
Spawn_Letter = str(Numbers_Corresponding_To_Building_Letters[str(random.randint(1,12))])
'''Write this down in a variable so that I can display it as well'''
Spawn_First_Or_Second_Building = str(random.randint(1,2))
'''Because there are so  many buildings in Henry's complex with the same letters, I have split it into either 1 or 2 this variable is randomly generated either 1 or 2'''
Spawn_Building_Number = 0

for key in All_Buildings_Dictionary:
    if Spawn_Letter == "A" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = "32"
        Spawn = All_Buildings[0]
        break
    if Spawn_Letter == "B" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number_With_Skips = str(random.randint(1, 2))
        if Spawn_Building_Number_With_Skips == "1":
            Spawn_Building_Number = str(random.randint(30, 31))
            if Spawn_Building_Number == "30":
                Spawn = All_Buildings[1]
                break
            if Spawn_Building_Number == "31":
                Spawn = All_Buildings[2]
                break
            break
        if Spawn_Building_Number_With_Skips == "2":
            Spawn_Building_Number = str(random.randint(35, 36))
            if Spawn_Building_Number == "35":
                Spawn = All_Buildings[3]
                break
            if Spawn_Building_Number == "36":
                Spawn = All_Buildings[4]
                break
            break
        break
    if Spawn_Letter == "B" and Spawn_First_Or_Second_Building == "2":
        Spawn_Building_Number = str(random.randint(35, 36))
        if Spawn_Building_Number == "35":
            Spawn = All_Buildings[5]
            break
        if Spawn_Building_Number == "36":
            Spawn = All_Buildings[6]
            break
        break
    if Spawn_Letter == "C" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(27, 28))
        if Spawn_Building_Number == "27":
            Spawn = All_Buildings[7]
            break
        if Spawn_Building_Number == "28":
            Spawn = All_Buildings[8]
            break
        break
    if Spawn_Letter == "E" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(26, 27))
        if Spawn_Building_Number == "26":
            Spawn = All_Buildings[9]
            break
        if Spawn_Building_Number == "27":
            Spawn = All_Buildings[10]
            break
        break
    if Spawn_Letter == "E" and Spawn_First_Or_Second_Building == "2":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[11]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[12]
            break
        break
    if Spawn_Letter == "F" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = "34"
        Spawn = All_Buildings[13]
        break
    if Spawn_Letter == "G" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = "29"
        Spawn = All_Buildings[14]
        break
    if Spawn_Letter == "H1" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 3))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[15]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[16]
            break
        if Spawn_Building_Number == "3":
            Spawn = All_Buildings[17]
            break
        break
    if Spawn_Letter == "H" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(6, 7))
        if Spawn_Building_Number == "6":
            Spawn = All_Buildings[18]
            break
        if Spawn_Building_Number == "7":
            Spawn = All_Buildings[19]
            break
        break
    if Spawn_Letter == "K" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 7))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[20]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[21]
            break
        if Spawn_Building_Number == "3":
            Spawn = All_Buildings[22]
            break
        if Spawn_Building_Number == "4":
            Spawn = All_Buildings[23]
            break
        if Spawn_Building_Number == "5":
            Spawn = All_Buildings[24]
            break
        if Spawn_Building_Number == "6":
            Spawn = All_Buildings[25]
            break
        if Spawn_Building_Number == "7":
            Spawn = All_Buildings[26]
            break
        break
    if Spawn_Letter == "L" and Spawn_First_Or_Second_Building == "1":
        Spawn = All_Buildings[27]
        Spawn_Building_Number = "1"
        break
    if Spawn_Letter == "M" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[28]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[29]
            break
        break
    if Spawn_Letter == "N" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[30]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[31]
            break
        break
    if Spawn_Letter == "N" and Spawn_First_Or_Second_Building == "2":
        Spawn_Building_Number = str(random.randint(1, 2))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[32]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[33]
            break
        break
    if Spawn_Letter == "P" and Spawn_First_Or_Second_Building == "1":
        Spawn_Building_Number = str(random.randint(1, 11))
        if Spawn_Building_Number == "1":
            Spawn = All_Buildings[34]
            break
        if Spawn_Building_Number == "2":
            Spawn = All_Buildings[35]
            break
        if Spawn_Building_Number == "3":
            Spawn = All_Buildings[36]
            break
        if Spawn_Building_Number == "4":
            Spawn = All_Buildings[37]
            break
        if Spawn_Building_Number == "5":
            Spawn = All_Buildings[38]
            break
        if Spawn_Building_Number == "6":
            Spawn = All_Buildings[39]
            break
        if Spawn_Building_Number == "7":
            Spawn = All_Buildings[40]
            break
        if Spawn_Building_Number == "8":
            Spawn = All_Buildings[41]
            break
        if Spawn_Building_Number == "9":
            Spawn = All_Buildings[42]
            break
        if Spawn_Building_Number == "10":
            Spawn = All_Buildings[43]
            break
        if Spawn_Building_Number == "11":
            Spawn = All_Buildings[44]
            break
        break
    break
print('Spawn Letter: "', str(Spawn_Letter), '", First Or Second Building: "', str(Spawn_First_Or_Second_Building), '" ', "Building Number: ", str(Spawn_Building_Number), sep="")
